src/explore-education-statistics-common/src/modules/table-tool/components/TableToolWizard.tsx

```diff
@@ -194,11 +197,27 @@ const TableToolWizard = ({
       locations,
       subjectId: state.query.subjectId,
     });
+    // Check if selected time period is in the time period options so can reset it if not.
+    const hasStartTimePeriod =
+      nextSubjectMeta.timePeriod.options.findIndex(
+        option =>
+          option.code === state.query.timePeriod?.startCode &&
+          option.year === state.query.timePeriod.startYear,
+      ) !== -1;
+    const hasEndTimePeriod =
+      nextSubjectMeta.timePeriod.options.findIndex(
+        option =>
+          option.code === state.query.timePeriod?.endCode &&
+          option.year === state.query.timePeriod.endYear,
+      ) !== -1;
 
     updateState(draft => {
       draft.subjectMeta.timePeriod = nextSubjectMeta.timePeriod;
-
       draft.query.locations = locations;
+      draft.query.timePeriod =
+        hasStartTimePeriod && hasEndTimePeriod
+          ? state.query.timePeriod
+          : undefined;
```

Comment:
How about if only one of the time periods exists in the meta? Would suggest that we still pre-populate the start/end date if it exists